sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -221,7 +221,14 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
             }
 
             val aggregateOperator =
-              if (functionsWithDistinct.isEmpty) {
+              if (aggregateExpressions.map(_.aggregateFunction).exists(!_.supportsPartial)) {
```

Comment:
I just noticed that if we have some thing like `select count(distinct a), hiveUDAF(b)...`, we will go this branch. So, we will silently ignore the distinct keyword.